### PR TITLE
Fix Pure Intentions

### PR DESCRIPTION
**Pull request by tool4ever**
Small part for #2193

### CODE REVIEW

**Comment by Hanmac** [comment]
are all ApiType.DealDamage updated, or is that just wishful thinking?

**Comment by tool4ever** [comment]
> are all ApiType.DealDamage updated, or is that just wishful thinking?

you mean DamageEach etc.? I left those alone because of the old PR and only touched this to get rid of the `NoTimetampCheck$` bandaid

if you mean all scripts, then no, but I can check those?
(my hope was because DamageDealEffect had timestamp check for years actually it should be fairy unproblematic 😬 )

**Comment by Hanmac** [comment]
> (my hope was because DamageDealEffect had timestamp check for years actually it should be fairy unproblematic 😬 )

the timestamp checks are only for the DamageSource property, but may not work for the DamageAmount (TriggeredCard) yet

See my other issue with Pyrotechnic Performer leaving the Battlefield

**Comment by tool4ever** [comment]
inside DamageDealEffect they're currently done for the affected (cards) which seems right
after more accuracy from game timestamps LKI fallback for NumDmg$ can be implemented?
(ideally inside AbilityUtils to server for all effects)

File: forge-game/src/main/java/forge/game/GameAction.java [hunk 1]
```diff
@@ -1911,8 +1911,6 @@ public final boolean destroy(final Card c, final SpellAbility sa, final boolean
             runParams.putAll(params);
         }
         game.getTriggerHandler().runTrigger(TriggerType.Destroyed, runParams, false);
-        // in case the destroyed card has such a trigger
-        game.getTriggerHandler().registerActiveLTBTrigger(c);
```
Comment (Hanmac) [inline forge-game/src/main/java/forge/game/GameAction.java:1915]: is the LTB Trigger registered somewhere deeper in the code?

Comment (Hanmac) [reply forge-game/src/main/java/forge/game/GameAction.java:1915]: then i think its okay?

**Pull request merged**